src/stream/src/executor/join/builder.rs

```diff
@@ -156,6 +157,29 @@ impl<const T: JoinTypePrimitive, const SIDE: SideTypePrimitive> JoinChunkBuilder
         }
     }
 
+    pub fn post_process(c: StreamChunk) -> StreamChunk {
+        let mut c = StreamChunkMut::from(c);
+
+        // NOTE(st1page): remove the pattern `UpdateDel(k, old), UpdateIns(k, NULL), UpdateDel(k, NULL),  UpdateIns(k, new)`
+        // to avoid this issue <https://github.com/risingwavelabs/risingwave/issues/17450>
+        let mut i = 1;
+        while i < c.capacity() {
+            if c.op(i - 1) == Op::UpdateInsert
+                && c.op(i) == Op::UpdateDelete
+                && c.row_ref(i) == c.row_ref(i - 1)
+            {
+                c.set_op(i - 2, Op::Delete);
+                c.set_vis(i - 1, false);
+                c.set_vis(i, false);
+                c.set_op(i + 1, Op::Insert);
+                i += 3;
+            } else {
+                i += 1;
+            }
+        }
+        c.into()
+    }
```

Comment:
A less intrusive and lower overhead approach, PTAL again when you have time @fuyufjh .